File: solutions/csharp/raindrops/2/Raindrops.cs
using System.Collections.Generic;
using System.Linq;

public static class Raindrops
{
    private static IEnumerable<(int, string)> _drops = new List<(int, string)>{
      (3, "Pling"),
      (5, "Plang"),
      (7, "Plong")
    };

    public static string Convert(int number)
    {
        var processed = _drops.Aggregate("", (s, i) => number % i.Item1 == 0 ? s + i.Item2 : s);
        return processed.Length == 0 ? number.ToString() : processed;
    }
}

File: solutions/fsharp/raindrops/2/Raindrops.fs
module Raindrops

let nameDrop divisor divisorName (number, fullName) =
    if number % divisor = 0 then
        (number, fullName + divisorName)
    else
        (number, fullName)

let unwrapDrop (number, fullName) =
    if String.length fullName > 0 then
        fullName
    else
        number.ToString()

let convert (number: int): string =
    (number, "")
    |> nameDrop 3 "Pling"
    |> nameDrop 5 "Plang"
    |> nameDrop 7 "Plong"
    |> unwrapDrop
